src/overlays/actors/ovl_Bg_Icefloe/z_bg_icefloe.c

```diff
@@ -27,35 +34,148 @@ const ActorInit Bg_Icefloe_InitVars = {
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyActor_Init(&this->dyna, 0);
+    DynaPolyActor_LoadMesh(globalCtx, &this->dyna, &object_icefloe_Colheader_000C90);
+    if (D_80AC5060 >= 3) {
+        s32 i;
+
+        if ((D_80AC4F30[0] != NULL) && (D_80AC4F30[0]->dyna.actor.update != NULL) &&
+            (D_80AC4F30[0]->actionFunc != func_80AC4D2C)) {
+            func_80AC4CF0(D_80AC4F30[0]);
+        }
+        for (i = 0; i < 2; i++) {
+            D_80AC4F30[i] = D_80AC4F30[i + 1];
+        }
+        D_80AC4F30[2] = this;
+    } else {
+        D_80AC4F30[D_80AC5060] = this;
+    }
+    D_80AC5060++;
+    this->dyna.actor.world.pos.y = this->dyna.actor.home.pos.y + 10.0f;
+    func_80AC4A80(this, globalCtx);
+}
+
+void BgIcefloe_Destroy(Actor* thisx, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    BgIcefloe* this = THIS;
+    s32 i;
+
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    D_80AC5060--;
+
+    for (i = 0; i < 3; i++) {
+        if (D_80AC4F30[i] == this) {
+            D_80AC4F30[i] = NULL;
+            break;
+        }
+    }
+}
+
+void func_80AC4A80(BgIcefloe* this, GlobalContext* globalCtx) {
+    this->unk_160 = 20;
+    SkinMatrix_Vec3fMtxFMultXYZW(&globalCtx->viewProjectionMtxF, &this->dyna.actor.world.pos,
+                                 &this->dyna.actor.projectedPos, &this->dyna.actor.projectedW);
+    Actor_PlaySfxAtPos(&this->dyna.actor, NA_SE_EV_ICE_STAND_APPEAR);
+    this->actionFunc = func_80AC4AE8;
+}
+
+static Vec3f sIceBlockAccel = { 0.0f, -0.5f, 0.0f };
+
+void func_80AC4AE8(BgIcefloe* this, GlobalContext* globalCtx) {
+    Vec3f velocity;
+    Vec3f position;
+
+    velocity.x = randPlusMinusPoint5Scaled(6.0f);
+    velocity.z = randPlusMinusPoint5Scaled(6.0f);
+    velocity.y = Rand_ZeroFloat(4.0f) + 4.0f;
+    this->dyna.actor.scale.x += 0.0064999997f;
+    this->dyna.actor.scale.z += 0.0064999997f;
+    this->dyna.actor.scale.y += 0.0064999997f;
+    position.x = this->dyna.actor.world.pos.x + (velocity.x * this->dyna.actor.scale.x * 75.0f);
+    position.z = this->dyna.actor.world.pos.z + (velocity.z * this->dyna.actor.scale.z * 75.0f);
+    position.y = this->dyna.actor.world.pos.y + (300.0f * this->dyna.actor.scale.y);
+    EffectSsIceBlock_Spawn(globalCtx, &position, &velocity, &sIceBlockAccel, Rand_S16Offset(10, 10));
+    this->unk_160--;
+    if (this->unk_160 == 0) {
+        func_80AC4C18(this);
+    }
+}
+
+void func_80AC4C18(BgIcefloe* this) {
+    this->unk_160 = this->dyna.actor.params;
+    this->actionFunc = func_80AC4C34;
+}
+
+void func_80AC4C34(BgIcefloe* this, GlobalContext* globalCtx) {
+    WaterBox* water;
+
+    this->unk_160--;
+    if ((this->unk_160 == 0) ||
+        (WaterBox_GetSurface1_2(globalCtx, &globalCtx->colCtx, this->dyna.actor.world.pos.x,
+                                this->dyna.actor.world.pos.z, &this->dyna.actor.home.pos.y, &water) == 0)) {
```

Comment:
```suggestion
        !WaterBox_GetSurface1_2(globalCtx, &globalCtx->colCtx, this->dyna.actor.world.pos.x,
                                this->dyna.actor.world.pos.z, &this->dyna.actor.home.pos.y, &water)) {
```
Looks like `WaterBox_GetSurface1_2` returns a boolean, so we can just use `!`.